undertow/src/main/java/org/wildfly/extension/undertow/ListenerAdd.java

```diff
@@ -105,7 +105,9 @@ public HttpHandler wrap(HttpHandler handler) {
         serviceBuilder.addDependency(workerServiceName, XnioWorker.class, service.getWorker())
                 .addDependency(socketBindingServiceName, SocketBinding.class, service.getBinding())
                 .addDependency(bufferPoolServiceName, (Injector) service.getBufferPool())
-                .addDependency(UndertowService.SERVER.append(serverName), Server.class, service.getServerService());
+                .addDependency(UndertowService.SERVER.append(serverName), Server.class, service.getServerService())
+                // Include the capability-declared service name as an alias  TODO get rid of the old style name
+                .addAliases(ListenerResourceDefinition.LISTENER_CAPABILITY.getCapabilityServiceName(name));
```

Comment:
I would do this bit differently.
in line 73 we have: 
`final ServiceName listenerServiceName = UndertowService.listenerName(name);`
which should be changed to 
`final ServiceName listenerServiceName = ListenerResourceDefinition.LISTENER_CAPABILITY.getCapabilityServiceName(name);` 

Unless your TODO was meant to do just that :-)